### ISSUE TITLE
Change distortion metric to MS-SSIM

### ISSUE THREAD

**Issue by chaohuang** [lyq133/LIC-HPCM#9]
The current code uses MSE as the distortion metric, and I wonder if there is a way to use MS-SSIM as the metric.

Thanks.

**Comment by lyq133** [comment]
Yes, it is possible to use MS-SSIM as the distortion metric. I have implemented a custom RateDistortionLoss class using the pytorch_msssim library.

```python
from pytorch_msssim import ms_ssim

class RateDistortionLoss(nn.Module):
    """Custom rate distortion loss with a Lagrangian parameter."""

    def __init__(self, lmbda=1e-2):
        super().__init__()
        self.ssim = ms_ssim
        self.lmbda = lmbda

    def forward(self, output, target):
        N, _, H, W = target.size()
        out = {}
        num_pixels = N * H * W

        out["bpp_loss"] = sum(
            (torch.log(likelihoods).sum() / (-math.log(2) * num_pixels))
            for likelihoods in output["likelihoods"].values()
        )
        out['y_bpp'] = torch.log(output['likelihoods']['y']).sum() / (-math.log(2) * num_pixels)
        out['z_bpp'] = torch.log(output['likelihoods']['z']).sum() / (-math.log(2) * num_pixels)
        out["ssim_loss"] = self.ssim(output["x_hat"], target, data_range=1.0)
        out["loss"] = self.lmbda * (1 - out["ssim_loss"]) + out["bpp_loss"]
        out["db_ssim"] = 10 * (torch.log(1 * 1 / (1 - out["ssim_loss"])) / np.log(10))

        return out
```
Hope this helps!
